src/include/sof/lib/uuid.h

```diff
@@ -10,6 +10,10 @@
 
 #include <sof/common.h>
 
+#ifdef __ZEPHYR__
```

Comment:
it would probably be better to abstract this into an arch macro

Reply:
That's largely what this is.  "`__ZEPHYR__`" is set in all C code compiled under a Zephyr app (via a setting in the top-level Zephyr CMakeLists.txt), it's used elsewhere in SOF as a tag for Zephyr-vs-legacy.  The usage here is an attempt to be minimal, basically each of Zephyr and XTOS define three small(ish) macros which are then used to implement the main UUID API.

Reply:
Yes, we have just been trying to abstract this into the "rtos" headers, this would be a "nice to have" but ill let others make the call on how much we want to be pristine here

Reply:
Ah, I get it.  You mean move the helper macros down into `{zephyr,xtos}/include/rtos/uuid.h` files and then just "`#include <rtos/uuid.h>`" here to get them?

That could be done, sure.  I guess my sense is that's more work than worth it for three macros in seven lines over two variants, but would be easy enough.

Reply:
yep exactly :)